src/shared/bpf-program.c

```diff
@@ -309,6 +309,13 @@ int bpf_map_new(enum bpf_map_type type, size_t key_size, size_t value_size, size
         attr.value_size = value_size;
         attr.max_entries = max_entries;
         attr.map_flags = flags;
+        strncpy(attr.map_name, name, sizeof(attr.map_name)-1);
+        for (int i = 0; attr.map_name[i]; i++) {
+                if (strchr("0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_.",
+                                        attr.map_name[i]) == NULL) {
+                        attr.map_name[i] = '_';
+                }
+        }
```

Comment:
sounds unnecessary to first copy the string and then mangle it. if you have to mangle, you might just copy it in the first place, i.e.:

```c
char *n = name;

for (size_t i = 0; i < sizeof(attr.map_name)-1 && *n != 0; i++, n++) 
        attr.map_name[i] = strchr(ALPHANUMERICAL ".", *n) ? *n : '_';
```

Reply:
maybe also add a comment here explaining the escaping further, i.e.: `/* the map name is primarily informational for debugging purposes, and typically too short to carry the full unit name, hence we employ a trivial lossy escaping + truncation to it, to make it fit */`

or something along those lines